### DISCUSSION TITLE
Packages built on GitHub actions windows doesn't work

### DISCUSSION THREAD

**Discussion by akshaybabloo** [community/community#34394] [Actions] [answered]
I am trying to build a Go asset in GitHub Action using the `windows-latest` image. After building it and downloading the asset, I get the following error on my PC:

```
$ git-alias -s tag
ResourceUnavailable: Program 'git-alias.exe' failed to run: An error occurred trying to start process 'C:\Users\gollaha\bin\git-alias.exe' with working directory 'C:\Users\gollaha\Code\personal\pretty-alias'. The specified executable is not a valid application for this OS platform.At line:1 char:1
```

I am not sure what is happening here, Linux and macOS seems to be working fine.

The repository can be found at [https://github.com/akshaybabloo/git-alias](https://github.com/akshaybabloo/git-alias)

**Comment by antonblr** [comment]
Try passing `GOOS` and `GOARCH` environment variables when building, e.g.:
```
GOOS=windows GOARCH=amd64 go build ...  # 64-bit
GOOS=windows GOARCH=386 go build ...    # 32 bit

# For MacOS
GOOS=darwin GOARCH=amd64 go build ...  # 64-bit  
GOOS=darwin GOARCH=386 go build ...    # 32-bit
```

**Reply by akshaybabloo** [reply, marked as answer]
I ended up using goreleaser. Thanks.
